web_ui/src/pages/create-project/components/pose-template/util.ts

```diff
@@ -118,6 +104,35 @@ export const updateWithLatestPoints =
         };
     };
 
-export const denormalizePoint = <T extends Point>(point: T, roi: RegionOfInterest): T => {
-    return { ...point, x: point.x * roi.width, y: point.y * roi.height };
+export const resizePoints = (scaleFactor: number, roi: RegionOfInterest, points: KeypointNode[]) => {
+    const [minX, maxX] = getMaxMinPoint(points, PointAxis.X);
+    const [minY, maxY] = getMaxMinPoint(points, PointAxis.Y);
+
+    const currentWidth = maxX - minX;
+    const currentHeight = maxY - minY;
+
+    const newWidth = currentWidth * scaleFactor;
+    const newHeight = currentHeight * scaleFactor;
+
+    const scaleX = newWidth / currentWidth;
+    const scaleY = newHeight / currentHeight;
```

Comment:
Do you need `newWidth` and `newHeight` here? I.e.
```
scaleX = newWidth / currentWidth = (currentWidth * scaleFactor) / currentWidth = scaleFactor
```

I guess below we do need it to add offset, but it might be simpler if we use `scaleFactor` directly instead of `scaleX` and `scaleY`?